Source/Urho3D/Audio/SoundSource.cpp

```diff
@@ -625,19 +646,35 @@ void SoundSource::MixMonoToMono(Sound* sound, int* dest, unsigned samples, int m
 
         if (sound->IsLooped())
         {
+            while (rampSamples--)
+            {
+                currentVolume += volDelta * (currentVolume != vol);
+                *dest = *dest + (*pos * currentVolume) / denom;
+                ++dest;
+                INC_POS_LOOPED();
+            }
+            if (!~rampSamples)
```

Comment:
This line (and also on other similar lines) looks weird to me. After the while loop immediately above it exits the block then the variable must become 0. Thus, that line always evaluates to `if (false)`. Anyway, I believe our linter also complaints about this one although not directly.

Reply:
No, if the while loop exits "normally" then the variable become
(unsigned)-1 (because of the postfix decrement), then the condition evaluated to true.
In this case it is just a safeguard, but the INC_POS_ONESHOT() macro can break the loop and in that case  the second loop will be skipped.